pkg/kubelet/volumemanager/cache/desired_state_of_world.go

```diff
@@ -203,7 +217,7 @@ func (dsw *desiredStateOfWorld) AddPodToVolume(
 	} else {
 		// For non-attachable volumes, generate a unique name based on the pod
 		// namespace and name and the name of the volume within the pod.
-		volumeName = volumehelper.GetUniqueVolumeNameForNonAttachableVolume(podName, volumePlugin, outerVolumeSpecName)
+		volumeName = volumehelper.GetUniqueVolumeNameForNonAttachableVolume(podName, volumePlugin, volumeSpec)
```

Comment:
nit: Double check this is a required change. Maybe we can pass in a dummy outerVolumeSpecName (same as volumeSpec.Name())?